src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(obj) {
+    if (obj instanceof Herbivore && !obj.hidden) {
+      obj.health -= 50;
+    }
+
+    const dead = Animal.alive.findIndex(beast => !beast.health);
+
+    if (dead !== -1) {
+      Animal.alive.splice(dead, 1);
+    }
```

Comment:
```suggestion
    const dead = Animal.alive.findIndex(beast => !beast.health);

    if (dead !== -1) {
      Animal.alive.splice(dead, 1);
    }
```
It's better to move this logic to the parent class